internal/pkg/config/v1/config.go

```diff
@@ -314,7 +388,7 @@ func (c *Config) Validate() error {
 		if _, ok := c.ContextStates[context.Name]; !ok {
 			c.ContextStates[context.Name] = new(ContextState)
 		}
-		if *c.ContextStates[context.Name] != *context.State {
+		if !c.IsTest && !reflect.DeepEqual(*c.ContextStates[context.Name], *context.State) {
```

Comment:
Later, we should write a PR that removes the need for `c.IsTest` and uses mocks. This is a bad habit to be getting into!